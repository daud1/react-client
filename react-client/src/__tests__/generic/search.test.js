import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Search from '../../components/generic/search.component';
import shallowToJson from 'enzyme-to-json';
import moxios from 'moxios';

describe ('Search Component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    
    it('matches snapshot', () => {
        const wrapper = shallow(<Search />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<Search />);

        wrapper.find('input[name="q"]').simulate('change', { target: { value: 'l' } });
        wrapper.find('form').simulate('submit', {preventDefault() {} });

        moxios.stubRequest('http://localhost:5000/shoppinglists/?q=l', {
            status: 200,
            'MSG': 'Success!'
        });
        moxios.wait(() => {
            expect(wrapper.html()).toContain('Success!');
            done();
        });
    });
})