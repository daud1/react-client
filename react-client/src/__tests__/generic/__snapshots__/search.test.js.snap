// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="q"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
